storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java

```diff
@@ -170,22 +170,41 @@ private Artifact _getArtifact(String artifactId, long version, EnumSet<ArtifactS
 
     private Artifact _getArtifact(String artifactId, EnumSet<ArtifactState> states) {
         requireNonNull(artifactId);
-        List<Artifact> artifacts = entityManager.createQuery(
-            "SELECT a FROM Artifact a " +
-            "WHERE a.artifactId = :artifact_id " +
-            "ORDER BY a.version DESC ", Artifact.class)
-                                                .setParameter("artifact_id", artifactId)
-                                                .getResultList();
-
-        for (Artifact artifact : artifacts) {
-            ArtifactState state = getState(artifactId, artifact.getVersion());
-            if (states == null || states.contains(state)) {
-                ArtifactStateExt.logIfDeprecated(artifactId, state, artifact.getVersion());
-                return artifact;
-            }
+        List<Artifact> artifacts;
+        if (states != null) {
+            String statesQuery = states
+                    .stream()
+                    .map(s -> s.toString())
+                    .collect(Collectors.joining(","));
+            String query = String.format(
+                     "SELECT a.* " +
+                     "FROM artifacts a " +
+                     "INNER JOIN " +
+                     "  (SELECT artifact_id, max(version) AS MaxVersion " +
+                     "  FROM meta " +
+                     "  WHERE artifact_id = '%s' AND key = '%s' AND value = ANY('{%s}'::text[]) " +
```

Comment:
No `ANY` operand is present on H2 database, that is the reason why tests are fails on CI
However, after changing to IN, the tests are still fails due to some logic errors